packages/core/src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,99 @@
+import pjson from "../../package.json";
+import { DEFAULT_TDK_DARKMATTER_BASE_URI } from "../constants";
+import { addCachedEvent, clearCachedEvents, getCachedEvents } from "./storage";
+import type { AnalyticsPayload, AppInfo, TrackableEvent } from "./types";
+import { getEventId, getServerTime } from "./utils";
+
+export class AnalyticsManager {
+  apiUri: string;
+
+  xApiKey: string;
```

Comment:
Thoughts on renaming to `apiKey`? I know the `X-` prefix is for the header, but doesn't really need to be in the instance var name. Not a big deal though